src/VirtoCommerce.CatalogModule.Data/Services/AssociationService.cs

```diff
@@ -72,53 +80,64 @@ public async Task SaveChangesAsync(IHasAssociations[] owners)
                     {
                         dbAssociation.ItemId = owner.Id;
                     }
+
                     changedEntities.AddRange(dbAssociations);
                 }
             }
 
-            await CreateOrUpdateAssociationAsync(changedEntities.ToArray());
+            using (var repository = _repositoryFactory())
+            {
+                var itemIds = owners.Where(x => x.Id != null).Select(x => x.Id).ToArray();
+                var existingEntities = repository.Associations.Where(x => itemIds.Contains(x.ItemId)).ToList();
+
+                var associationComparer = AnonymousComparer.Create((AssociationEntity x) => x.ItemId + ":" + x.AssociationType + ":" + x.AssociatedItemId + ":" + x.AssociatedCategoryId);
```

Comment:
this code will only update association without creating a new or delete in the DB. It confuses with the methods comments.

Reply:
maybe better to use the more compact form?
```C#
existingEntities.ObserveCollection(x=> repository.add(x), x=> repository.remove(x));
changedEntities.Patch(existingEntities, associationComparer, (sourceAssociation, targetAssociation) => sourceAssociation.Patch(targetAssociation), repository);

```

Reply:
Done